app/app/components/Profile.jsx

```diff
@@ -33,81 +35,117 @@ const Profile = ({ userId, booksReadCount }) => {
 
     const fetchFavoriteGenre = async () => {
       try {
-        const response = await axios.get(
-          `${process.env.NEXT_PUBLIC_API_BASE_URL}/api/user-books/favorite-genre`,
-          {
-            withCredentials: true,
-          }
+        const response = await makeRequest(
+          "/api/user-books/favorite-genre",
+          {},
+          "GET"
         );
-        setFavoriteGenre(response.data.favoriteGenre);
+        setFavoriteGenre(response.favoriteGenre);
       } catch (err) {
         console.error("Error fetching favorite genre:", err);
         setFavoriteGenre("Not available.");
       }
     };
 
+    const fetchActiveGoal = async () => {
+      try {
+        const response = await makeRequest(`/api/goals`, {}, "GET");
+        const goalsArray = response.goals;
+
+        if (Array.isArray(goalsArray)) {
+          const active = goalsArray.find((g) => g.status === "IN_PROGRESS");
+          setActiveGoal(active || null);
+        } else {
+          console.error("Goals data is not an array:", response);
+        }
+      } catch (error) {
+        console.error("Failed to fetch goals:", error);
+      }
+    };
+
     fetchUserProfile();
     fetchFavoriteGenre();
-  }, [userId]);
+    fetchActiveGoal();
+  }, []);
+
+  // Memoize updateBooksReadCount to prevent unnecessary re-renders
+  const updateBooksReadCount = useCallback(
```

Comment:
I don't quite see where is that used, but this could be because of the github diff. From what I see though, the function only uses a state setter inside so it does not need to be memoized - the maintenance of that is handled by the library